### PR TITLE
Reworked HackerEarth scraper

### PR DESCRIPTION
**Pull request by LaRuim**
HackerEarth scraper has similarly been reworked to be functional; Bug fixes for CodeChef scraper
HackerEarth scraper also works in similar fashion to the CodeChef scraper in that it takes a list of contest ID's; 
It finds the ICPC id on the webpage of the contest, send a request to the API via the ICPC id, and parses it.

### CODE REVIEW

File: scrapers/codechef.py [hunk 1]
```diff
@@ -70,11 +71,10 @@ def scrape(contest_codes):
             easy_points=100000 # Initial value set to points per problem
 
         for Division in divisions: # Build the scraped scoreboard
-            divisions[Division].problems = get_problems(f"{site_url}/{contest_code}{Division}")
-            divisions[Division].scraped_scoreboard = get_rankings(f"{scoreboard_base_url}/{contest_code}{Division}{scoreboard_filter_query}", contest_code)
+            divisions[Division] = divisions[Division]._replace(problems = get_problems(f"{site_url}/{contest_code}{Division}"))
```
Comment (souravtecken) [inline scrapers/codechef.py:74]: Oops

**Review by souravtecken — COMMENTED** [review]
@LaRuim, Have you tested the scrapers after reworking them?

**Comment by LaRuim** [comment]
Yes; both work fine.

Codechef, the reason it bugged out was because I'd accepted a change you made without realizing.

You've used a namedtuple, but tis immutable; when you changed the functions init_site and init_rankings to get_site and get_rankings, you were tryna mutate the namedtuple.

I didn't realize I accepted that change, so I didn't test it, my bad; I shall retest after every incoming change from now

**Comment by souravtecken** [comment]
> Yes; both work fine.
> 
> Codechef, the reason it bugged out was because I'd accepted a change you made without realizing.
> 
> You've used a namedtuple, but tis immutable; when you changed the functions init_site and init_rankings to get_site and get_rankings, you were tryna mutate the namedtuple.
> 
> I didn't realize I accepted that change, so I didn't test it, my bad; I shall retest after every incoming change from now

Realized, that's why my "Oops" above

**Pull request merged**